hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -287,15 +277,25 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
               e);
         }
 
-        LOG.error(
-            "Cannot tell if commit to {}.{} succeeded, attempting to reconnect and check.",
-            database,
-            tableName,
-            e);
-        commitStatus = BaseMetastoreOperations.CommitStatus.UNKNOWN;
-        commitStatus =
-            BaseMetastoreOperations.CommitStatus.valueOf(
-                checkCommitStatus(newMetadataLocation, metadata).name());
+        if (e.getMessage() != null
+            && e.getMessage()
```

Comment:
Could we have a same issue here than with #11576?
If there is a new exception during the `handleConcurrentModification` call, then the commit status is unset, and the metadata will be removed.

Maybe leave the
```
commitStatus = BaseMetastoreOperations.CommitStatus.UNKNOWN;
```
line before the `if` statement?

Reply:
Also, we could add a comment there. It took me a while the understand the reasoning behind it 😄

Reply:
Good catch, will do